nexus-repository-conan-it/src/test/java/org/sonatype/nexus/plugins/conan/internal/ConanRoutingRuleIT.java

```diff
@@ -0,0 +1,131 @@
+
+  private Repository proxyRepo;
+
+  private Server server;
+
+  private static final String DOWNLOAD_URLS_FILE_NAME = "download_urls";
+
+  private static final String JSON_FOR_MODERN_CPP_URL =
+      "v1/conans/jsonformoderncpp/3.7.0/vthiery/stable/packages/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/" +
+          DOWNLOAD_URLS_FILE_NAME;
+
+  private static final String JSON_FOR_MODERN_CPP_BLOCKED_URL =
+      "v1/conans/jsonformoderncpp/3.7.0/blockedvendor/stable/packages/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/" +
+          DOWNLOAD_URLS_FILE_NAME;
+
+  private static final String BLOCKED_VENDOR_REGEX = ".*/blockedvendor/.*";
+
+  @Inject
+  private RoutingRuleStore ruleStore;
+
+  @Rule
+  public RepositoryRuleConan repos = new RepositoryRuleConan(() -> repositoryManager);
+
+  @Rule
+  public RoutingRuleRule routingRules = new RoutingRuleRule(() -> ruleStore);
+
+  private EntityId createBlockedRoutingRule(final String name, final String matcher) {
+    RoutingRule rule = routingRules.create(name, matcher);
+    return rule.id();
+  }
+
+  private void attachRuleToRepository(final Repository repository, final EntityId routingRuleId) throws Exception {
+    org.sonatype.nexus.repository.config.Configuration configuration = repository.getConfiguration();
+    configuration.setRoutingRuleId(routingRuleId);
+    repositoryManager.update(configuration);
+  }
+
+  @Configuration
+  public static Option[] configureNexus() {
+    return NexusPaxExamSupport.options(
+        NexusITSupport.configureNexusBase(),
+        nexusFeature("org.sonatype.nexus.plugins", "nexus-repository-conan")
+    );
+  }
+
+  @Before
+  public void setup() throws Exception {
+    server = Server.withPort(0).serve("/" + JSON_FOR_MODERN_CPP_URL)
+        .withBehaviours(Behaviours.file(testData.resolveFile(DOWNLOAD_URLS_FILE_NAME)))
+        .serve("/" + JSON_FOR_MODERN_CPP_BLOCKED_URL)
+        .withBehaviours(Behaviours.file(testData.resolveFile(DOWNLOAD_URLS_FILE_NAME)))
+        .start();
+
+    proxyRepo = repos.createConanProxy(testName.getMethodName(), server.getUrl().toExternalForm());
+    proxyClient = conanClient(proxyRepo);
+    EntityId routingRuleId = createBlockedRoutingRule("conan-rule", BLOCKED_VENDOR_REGEX);
+    attachRuleToRepository(proxyRepo, routingRuleId);
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    server.stop();
+  }
+
+  private void assertGetResponseStatus(
+      final ConanClient client,
+      final Repository repository,
+      final String path,
+      final int responseCode) throws IOException
+  {
+    try (CloseableHttpResponse response = client.get(path)) {
+      StatusLine statusLine = response.getStatusLine();
+      assertThat("Repository:" + repository.getName() + " Path:" + path, statusLine.getStatusCode(), is(responseCode));
+    }
+  }
+
+  @Test
+  public void testRoutingRule() throws Exception {
+    assertGetResponseStatus(proxyClient, proxyRepo, JSON_FOR_MODERN_CPP_URL, 200);
```

Comment:
Maybe use javax.ws.rs.core.Response.Status.OK.getStatusCode()? and for 403 below?

Reply:
574f250fe13f0b7c107dce9df11b50e597f7bcaf 
Is it better now?

Reply:
I think yes